File: wrappers/matlab/ppr_path_test.m
A=readSMAT('../../graph/usps_3nn_list.smat');
seedids=[7575];
nseedids=1;
alpha=0.99;
eps=0.0001;
rho=0.0;
[xlength,~]=size(A);
[actual_length,xids]=ppr_path(A,seedids,nseedids, ...
                              alpha,eps,rho,xlength);
actual_length
xids'

File: wrappers/python/ppr_path_test.py
from ppr_path import ppr_path
from list_to_CSR import list_to_CSR
import numpy as np

(m,n,ai,aj) = list_to_CSR("../../graph/usps_3nn.smat")
alpha=0.99
eps=10**(-4)
rho=0.0
xlength=n
f = open("../../graph/usps_3nn_seed.smat")
data = f.read()
data = data.split()
nseedids = int(data[0])
seedids = []
for i in range(nseedids):
    seedids += [data[i + 1]]
f.close()
(actual_length,xids) = ppr_path(n,ai,aj,alpha,eps,rho,seedids,nseedids,xlength)
print actual_length,xids
